src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -113,102 +145,35 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
-            _ <- processVotesForBlocks(blocksWithAccounts.map{case (block, _) => block}) andThen logVotingOutcome
-          } yield blocksWithAccounts.size
-
-      }
-
-    blockPagesToSynchronize.flatMap {
-      // Fails the whole process if any page processing fails
-      case (pages, total) => Future.fromTry(Try {
-        //custom progress tracking for blocks
-        val logProgress = logProcessingProgress(entityName = "block", totalToProcess = total, processStartNanos = System.nanoTime()) _
-
-        /* Process in a strictly sequential way, to avoid opening an unbounded number of requests on the http-client.
-        * The size and partition of results is driven by the NodeOperator itself, where each page contains a
-        * "thunked" future of the result.
-        * Such future will be actually started only as the page iterator is scanned, one element at the time
-        */
-        pages.foldLeft(0) {
-          (processed, nextPage) =>
-            //wait for each page to load, before looking at the next, thus  starting the new computation
-            val justDone = Await.result(
-              processBlocksPage(nextPage) <* processTezosAccounts(),
-              atMost = batchingConf.blockPageProcessingTimeout)
-            processed + justDone <| logProgress
-        }
-      })
-    } transform {
-      case Failure(e) =>
-        val error = "Could not fetch blocks from client"
-        logger.error(error, e)
-        Failure(BlocksProcessingFailed(message = error, e))
-      case Success(_) => Success(Done)
-    }
-
-  }
-
-  /**
-    * Fetches voting data for the blocks and stores any relevant
-    * result into the appropriate database table
-    */
-  def processVotesForBlocks(blocks: List[TezosTypes.Block]): Future[Option[Int]] = {
-    import cats.syntax.traverse._
-    import cats.syntax.foldable._
-    import cats.syntax.semigroup._
-    import cats.instances.list._
-    import cats.instances.option._
-    import cats.instances.int._
-    import slickeffect.implicits._
-
-
-    tezosNodeOperator.getVotingDetails(blocks).flatMap {
-      case (proposals, bakersBlocks, ballotsBlocks) =>
+  * Fetches all blocks not in the database from the Tezos network and adds them to the database.
+  * Additionally stores account references that needs updating, too
+  */
+  private[this] def processBlocks(node: NodeOperator): Future[Done] = {
+
+    /* additional voting data fetching */
+    def processVotes(blocks: List[Block]): Future[Option[Int]] = {
+      import cats.syntax.traverse._
+      import cats.syntax.foldable._
+      import cats.syntax.semigroup._
+      import cats.instances.list._
+      import cats.instances.option._
+      import cats.instances.int._
+      import slickeffect.implicits._
+
+      node.getVotingDetails[Future](blocks).flatMap {
+        case (proposals, bakersBlocks, ballotsBlocks) =>
```

Comment:
writeVotingProposals and writeVotingBakers currently have errors saying "PostgresProfile.api.DBIO[Option[Int]] doesn't conform to the expected type G_[B_]"

Reply:
weird, I'll possibly disregard these errors as I will publish the final version, so this actual PR will never be actually merged...

Right now I don't even know how hard it is to apply current master changes to the WIP branch 😛